src/components/MAPI/clusters/ClusterDetail/ClusterDetailActions.tsx

```diff
@@ -0,0 +1,222 @@
+  >(clusterKey, () =>
+    fetchCluster(clusterClient.current, auth, provider, namespace!, clusterId)
+  );
+
+  const {
+    data: nodePoolList,
+    error: nodePoolListError,
+  } = useSWR(fetchNodePoolListForClusterKey(cluster), () =>
+    fetchNodePoolListForCluster(clientFactory, auth, cluster)
+  );
+
+  useEffect(() => {
+    if (nodePoolListError) {
+      ErrorReporter.getInstance().notify(nodePoolListError);
+
+      const errorMessage = extractErrorMessage(nodePoolListError);
+      new FlashMessage(
+        `There was a problem loading node pools for cluster <code>${clusterId}</code>.`,
+        messageType.ERROR,
+        messageTTL.FOREVER,
+        errorMessage
+      );
+    }
+  }, [nodePoolListError, clusterId]);
+
+  const appListClient = useRef(clientFactory());
+  const appListGetOptions = { namespace: clusterId };
+  const { data: appList, error: appListError } = useSWR<
+    applicationv1alpha1.IAppList,
+    GenericResponseError
+  >(applicationv1alpha1.getAppListKey(appListGetOptions), () =>
+    applicationv1alpha1.getAppList(
+      appListClient.current,
+      auth,
+      appListGetOptions
+    )
+  );
+
+  useEffect(() => {
+    if (appListError) {
+      ErrorReporter.getInstance().notify(appListError);
+
+      const errorMessage = extractErrorMessage(appListError);
+      new FlashMessage(
+        `There was a problem loading apps for cluster <code>${clusterId}</code>.`,
+        messageType.ERROR,
+        messageTTL.FOREVER,
+        errorMessage
+      );
+    }
+  }, [appListError, clusterId]);
+
+  const userInstalledApps = useMemo(() => {
+    if (!appList) return undefined;
+
+    return filterUserInstalledApps(appList.items, true);
+  }, [appList]);
+
+  const hasError =
+    typeof orgError !== 'undefined' ||
+    typeof clusterError !== 'undefined' ||
+    typeof nodePoolListError !== 'undefined';
+
+  const isLoadingResources =
+    typeof org === 'undefined' &&
+    typeof orgError === 'undefined' &&
+    typeof cluster === 'undefined' &&
+    typeof clusterError === 'undefined' &&
+    typeof nodePoolList === 'undefined' &&
+    typeof nodePoolListError === 'undefined' &&
+    typeof appList === 'undefined' &&
+    typeof appListError === 'undefined';
+
+  const [isDeleting, setIsDeleting] = useState(false);
+
+  const handleDelete = async () => {
+    if (!cluster) return;
+
+    setIsDeleting(true);
+
+    try {
+      const client = clientFactory();
+
+      await deleteCluster(client, auth, cluster);
+
+      // Success message and redirection are handled by the parent component.
+    } catch (err) {
+      ErrorReporter.getInstance().notify(err);
+
+      const errorMessage = extractErrorMessage(err);
+
+      new FlashMessage(
+        `Could not delete cluster <code>${cluster.metadata.name}</code>:`,
+        messageType.ERROR,
+        messageTTL.LONG,
+        errorMessage
+      );
+
+      setIsDeleting(false);
+    }
+  };
+
+  const isLoading = isLoadingResources || isDeleting;
```

Comment:
If you come from the `Overview` tab, then you won't see any loading animations at first, because all the data is already cached. However, if you visit the `/actions` page directly, the deletion button will be disabled and you will see a loading animation.